tests/auxiliary/minicore.rs

```diff
@@ -101,10 +156,70 @@ macro_rules! concat {
         /* compiler built-in */
     };
 }
+
 #[rustc_builtin_macro]
 #[macro_export]
 macro_rules! stringify {
     ($($t:tt)*) => {
         /* compiler built-in */
     };
 }
+
+#[macro_export]
+macro_rules! panic {
+    ($msg:literal) => {
+        $crate::panic(&$msg)
+    };
+}
+
+#[rustc_intrinsic]
+#[rustc_intrinsic_const_stable_indirect]
+#[rustc_intrinsic_must_be_overridden]
+pub const fn size_of<T>() -> usize {
+    loop {}
+}
+
+#[rustc_intrinsic]
+#[rustc_intrinsic_must_be_overridden]
+pub const fn abort() -> ! {
+    loop {}
+}
+
+#[lang = "panic"]
+#[rustc_const_panic_str]
+const fn panic(_expr: &&'static str) -> ! {
+    abort();
+}
+
+#[lang = "eq"]
+pub trait PartialEq<Rhs: ?Sized = Self> {
+    fn eq(&self, other: &Rhs) -> bool;
+    fn ne(&self, other: &Rhs) -> bool {
+        !self.eq(other)
+    }
+}
+
+impl PartialEq for usize {
+    fn eq(&self, other: &usize) -> bool {
+        (*self) == (*other)
+    }
+}
```

Comment:
Suggestion: I think you could use a macro to construct these `impl PartialEq`s so as long as the shape remains compatible since I think that's also what `core` does.